lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -200,4 +220,21 @@ def self.import_options(git_repo, options)
   end
 
   private_class_method :import_options
+
+  def self.validate_refs(repo, options)
+    match = nil
+    case options['ref_type'].downcase
```

Comment:
At a minimum, the outermost method should downcase ref_type and then it's just that way all the ay down so you don't have to keep tweaking it in every method.

Additionally, this validate_refs method should probably be private, so you don't have to worry about other things getting into the method.